content/concepts/core-concepts.md

```diff
@@ -2,12 +2,31 @@
 title: Core Concepts
 ---
 
-## Entry, Output, Loaders, Plugins
+## Overview
```

Comment:
I would say do an article per concept so you have `concepts/entry.md`, `concepts/output.md`, and so on. `concepts/index.md` can aggregate these and provide very short explanations and literally work as an index. There's a bit of explanation already. This would be in addition to that I think.